src/Config/Parser/AnnotationParser.php

```diff
@@ -452,26 +424,43 @@ private static function enumAnnotationToGQLConfiguration(GQL\Enum $enumAnnotatio
         }
 
         $enumConfiguration = ['values' => $values];
-        $enumConfiguration = self::getDescriptionConfiguration($classAnnotations) + $enumConfiguration;
+        $enumConfiguration = self::getDescriptionConfiguration($graphClass->getAnnotations()) + $enumConfiguration;
 
         return ['type' => 'enum', 'config' => $enumConfiguration];
     }
 
     /**
      * Get a GraphQL Union configuration from given union annotation.
      */
-    private static function unionAnnotationToGQLConfiguration(string $className, GQL\Union $unionAnnotation, array $classAnnotations, array $methods): array
+    private static function unionAnnotationToGQLConfiguration(GraphClass $graphClass, GQL\Union $unionAnnotation): array
     {
-        $unionConfiguration = ['types' => $unionAnnotation->types];
-        $unionConfiguration = self::getDescriptionConfiguration($classAnnotations) + $unionConfiguration;
+        $unionConfiguration = [];
+        if ($unionAnnotation->types) {
```

Comment:
Shouldn't it be here `!empty($unionAnnotation->types)` ?

I would generally suggest to use more specific comparison everywhere. It is more save and predictable